decoder/line_reader.go

```diff
@@ -47,18 +47,27 @@ func NewLineReader(reader io.Reader, maxLineLength int) *LineReader {
 // return the first `maxLineLength` bytes with `ErrLineTooLong`. On the next
 // call it will return the next line.
 func (lr *LineReader) ReadLine() ([]byte, error) {
+	var prefix bool
 	for {
-		line, prefix, err := lr.br.ReadLine()
-		if err != nil {
-			return nil, err
+		line, err := lr.br.ReadSlice('\n')
+		if err == bufio.ErrBufferFull {
+			prefix = true
+		} else if err == io.EOF && lr.skip {
+			lr.skip = false
+			return line, ErrLineTooLong
```

Comment:
This will return the remainder of a too-long line on io.EOF -- why return it on EOF when we don't in other cases? I *think* this special case should be moved below, so it's like:

```go
if !lr.skip {
   ...
} else if err == io.EOF {
    return nil, io.EOF
} else if !prefix {
    lr.skip = false
}
```